src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/Information.vb

```diff
@@ -0,0 +1,384 @@
+            If VarName Is Nothing Then
+                Return VariantType.Object
+            End If
+
+            Return VarTypeFromComType(VarName.GetType())
+        End Function
+
+        Friend Function VarTypeFromComType(ByVal typ As System.Type) As VariantType
+            If typ Is Nothing Then
+                Return VariantType.Object
+            End If
+
+            If typ.IsArray() Then
+
+                typ = typ.GetElementType()
+                If typ.IsArray Then
+                    Return CType(VariantType.Array Or VariantType.Object, VariantType)
+                End If
+
+                Dim result As VariantType = VarTypeFromComType(typ)
+                If (result And VariantType.Array) <> 0 Then
+                    'Element type is also an array, so just return "array of objects"
+                    Return CType(VariantType.Array Or VariantType.Object, VariantType)
+                End If
+                Return CType(result Or VariantType.Array, VariantType)
+
+            ElseIf typ.IsEnum() Then
+                typ = System.Enum.GetUnderlyingType(typ)
+            End If
+
+            If typ Is Nothing Then
+                Return VariantType.Empty
+            End If
+
+            Select Case Type.GetTypeCode(typ)
+
+                Case TypeCode.String
+                    Return VariantType.String
+                Case TypeCode.Int32
+                    Return VariantType.Integer
+                Case TypeCode.Int16
+                    Return VariantType.Short
+                Case TypeCode.Int64
+                    Return VariantType.Long
+                Case TypeCode.Single
+                    Return VariantType.Single
+                Case TypeCode.Double
+                    Return VariantType.Double
+                Case TypeCode.DateTime
+                    Return VariantType.Date
+                Case TypeCode.Boolean
+                    Return VariantType.Boolean
+                Case TypeCode.Decimal
+                    Return VariantType.Decimal
+                Case TypeCode.Byte
+                    Return VariantType.Byte
+                Case TypeCode.Char
+                    Return VariantType.Char
+                Case TypeCode.DBNull
+                    Return VariantType.Null
+
+            End Select
+
+            If (typ Is GetType(System.Reflection.Missing)) OrElse
+               (typ Is GetType(System.Exception)) OrElse
+               (typ.IsSubclassOf(GetType(System.Exception))) Then
+                Return VariantType.Error
+            ElseIf typ.IsValueType() Then
+                Return VariantType.UserDefinedType
+            Else
+                Return VariantType.Object
+            End If
+
+        End Function
+
+#Region " BACKWARDS COMPATIBILITY.  These functions (IsNumeric, TypeName, SystemTypeName, VbTypeName) have been superceded by the versions in Versioned.vb "
+
+        'WARNING WARNING WARNING WARNING WARNING
```

Comment:
I think this warning can be removed (Everett = VS2003, Orcas = VS2005...)

Also the region markers can go - there is no need to record compat with VS2002. Also I notice that in MSDN, Versioned.IsNumeric is marked not to use, so maybe it's not so superseded.

Reply:
I had never heard of the Versioned namespace. What appears to happen is that the VB compiler silently moves calls to Information.IsNumeric to Versioned.IsNumeric. So the Information variants exist for binary backwards compatibility (or calls from other languages).